### ISSUE TITLE
ACLDatacenters with upper-case characters breaks ACL support

### ISSUE THREAD

**Issue by joelthompson** [hashicorp/consul#1778]
Let's say I have a test Consul config:

``` json
$ cat consul-config/config.json 
{
  "server": true,
  "leave_on_terminate": false,
  "log_level": "DEBUG",
  "disable_remote_exec": true,
  "rejoin_after_leave": true,
  "skip_leave_on_interrupt": true,
  "log_level": "DEBUG",
  "bootstrap_expect": 1,
  "acl_default_policy": "allow",
  "acl_master_token": "asdfasdf",
  "datacenter": "Test",
  "acl_datacenter": "Test",
  "data_dir": "/mnt/consul"
}
```

in which the datacenter and acl_datacenter have an uppercase letter. If I start Consul using this config, `$ ./consul agent -server -config-dir ./consul-config/ -data-dir /tmp/consul/ -advertise 127.0.0.1` then this shows up in the output:

>    2016/03/01 17:06:24 [WARN] consul.rpc: RPC request for DC 'Test', no path found
>    2016/03/01 17:06:24 [ERR] consul.acl: Failed to get policy for 'anonymous': No path to datacenter
>    2016/03/01 17:06:24 [WARN] consul.rpc: RPC request for DC 'Test', no path found
>    2016/03/01 17:06:24 [ERR] consul.acl: Failed to get policy for 'anonymous': No path to datacenter

And if I try to do any operations against ACLs, e.g., `curl http://localhost:8500/v1/acl/list` it returns `No path to datacenter` and this shows up in the Consul output:

> ```
> 2016/03/01 17:06:54 [WARN] consul.rpc: RPC request for DC 'Test', no path found
> 2016/03/01 17:06:54 [ERR] http: Request GET /v1/acl/info/anonymous, error: No path to datacenter from=127.0.0.1:34349
> ```
> 
>    2016/03/01 17:06:54 [DEBUG] http: Request GET /v1/acl/info/anonymous (199.125µs) from=127.0.0.1:34349

I think the problem is that command.go does a ToLower on config.Datacenter, but it doesn't do a ToLower on config.ACLDatacenter, so internally Consul thinks there's only a datacenter named `test` but it's looking for the datacenter `Test` with a capital T to validate ACLs but that doesn't exist.

**Comment by perzizzle** [comment]
I am also able to reproduce this issue.

**Comment by mynameisgv** [comment]
Thanks. After many tries, this helped me understand what is wrong.
I am also able to reproduce this issue and replacing datacenter to lowercase fixed the issue.

**Comment by kyhavlov** [comment]
Fixed in #2478
